habitat_baselines/il/trainers/edfe_trainer.py

```diff
@@ -0,0 +1,249 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import time
+
+import numpy as np
+import torch
+from torch.utils.data import DataLoader
+
+import habitat
+from habitat import logger
+from habitat_baselines.common.base_trainer import BaseILTrainer
+from habitat_baselines.common.baseline_registry import baseline_registry
+from habitat_baselines.common.tensorboard_utils import TensorboardWriter
+from habitat_baselines.il.data.edfe_data import EDFEDataset
+from habitat_baselines.il.models.models import MultitaskCNNOutput
+
+
+@baseline_registry.register_trainer(name="edfe")
+class EDFETrainer(BaseILTrainer):
+    r"""Trainer class for Encoder-Decoder for Feature Extraction
+    used in EmbodiedQA (Das et. al.;CVPR 2018)
+    Paper: https://embodiedqa.org/paper.pdf.
+    """
+    supported_tasks = ["EQA-v0"]
+
+    def __init__(self, config=None):
+        super().__init__(config)
+
+        if config is not None:
+            logger.info(f"config: {config}")
+
+    def train(self) -> None:
+        r"""Main method for training Navigation model of EQA.
+
+        Returns:
+            None
+        """
+        config = self.config
+
+        assert torch.cuda.is_available(), "Cuda-enabled GPU required"
+        torch.cuda.set_device(config.TORCH_GPU_ID)
+
+        env = habitat.Env(config=config.TASK_CONFIG)
+
+        edfe_dataset = EDFEDataset(env, config)
+
+        train_loader = DataLoader(
+            edfe_dataset, batch_size=config.IL.EDFE.batch_size, shuffle=True
+        )
+
+        logger.info("train_loader has %d samples" % len(edfe_dataset))
+
+        model = MultitaskCNNOutput(num_classes=41)
+        model.double().train().cuda()
```

Comment:
Just saw this -- why are you doing everything in double precision?  Consumer GPUs generally don't support double precision well and are really slow with it.

Reply:
@erikwijmans Thank you for pointing this out. The model's input happened to be in double precision, which raised an error about mismatch of weight type (float) and input type (double). Instead of changing the input data type, I had casted the model to double type. 

I have made the suggested changes and removed the conversion to double in the latest commit. Thank you